### ISSUE TITLE
Access to German phone book Klicktel

### ISSUE THREAD

**Issue by jammerhexe** [nvaccess/nvda#6135]
The most used phone book CD by blind People in Germany is the CD from Klicktel. Unfortunately the lists and menues of the database isn't accessible with NVDA. I could provide an ISO Image of an issue of Klicktel from last year via Dropbox.
Because NVDA reads the first entry in the list it may be only a minor Problem to add Access to this Software.

**Comment by jammerhexe** [comment]
[> https://dl.dropboxusercontent.com/u/35365392/Klicktel.iurl)

**Comment by jammerhexe** [comment]
Forgot to add the link to my Dropbox
https://dl.dropboxusercontent.com/u/35365392/Klicktel.iso

**Comment by Adriani90** [comment]
@jammerhexe are you still able to reproduce this issue? Is it working with NVDA 2019.1 Beta 1?

**Comment by jammerhexe** [comment]
Yes, Klicktel still isn‘t accessible. The iso image of the old version should exist on my nas



This is an I-Mail

**Comment by Adriani90** [comment]
Ok so you tested with last clicktel version. Right?

**Comment by jammerhexe** [comment]
No, as you know Klicktel comes two times a year.  My issue is from spring 2018. but I suppose they haven‘t changed someting of the osm since many year years


This is an I-Mail

**Comment by Adriani90** [comment]
This is still an issue with clicktel 2024, NVDA cannot really read anything on its interface except for confirmation dialogs and some of the edit fields. With object navigation it is possible to explore some of the controls, though most of them do not have any accessible labels and you cannot route the mouse to them.
Jaws doesn't work with this interface at all, its virtual jaws cursor can read some of the controls but it is not possible to work with it.
Narator seems to be the only screen reader that recognizes properly some of the controls via its scan mode, but it is still far from an accessible UX.
The menu and the toolbar are not accessible at all with any of the screen readers.

We didn't read any user requests for clicktel accessibility since many years, neither in the german communication channels. It might be better to contact the company directly and to start an advisory initiative if they want. There is nothing NVDA can do at this stage.

I am closing this particular issue as can't fix.

**Comment by jammerhexe** [comment]
Hi,

my old version 11 of COBRA (last version before Baum wend out of business) still have no problems with Klicktel. Anything needed is accessible with COBRA.
So it seems it isn’t impossible to solve the problem at all.

Yours sincerely

Arno


Von: Adriani90
Gesendet: Samstag, 6. April 2024 23:25
An: nvaccess/nvda
Cc: jammerhexe; Mention
Betreff: Re: [nvaccess/nvda] Access to German phone book Klicktel (#6135)

This is still an issue with clicktel 2024, NVDA cannot really read anything on its interface except for confirmation dialogs and some of the edit fields. With object navigation it is possible to explore some of the controls, though most of them do not have any accessible labels and you cannot route the mouse to them.
Jaws doesn't work with this interface at all, its virtual jaws cursor can read some of the controls but it is not possible to work with it.
Narator seems to be the only screen reader that recognizes properly some of the controls via its scan mode, but it is still far from an accessible UX.
The menu and the toolbar are not accessible at all with any of the screen readers.
We didn't read any user requests for clicktel accessibility since many years, neither in the german communication channels. It might be better to contact the company directly and to start an advisory initiative if they want. There is nothing NVDA can do at this stage.
I am closing this particular issue as can't fix.

**Comment by jammerhexe** [comment]
Hi Adriani,

in addition, if you didn’t run COBRA – there is a download site of the German Screenreader mailing list at 
http://screenreader.downloads.blindzeln.org
user and passwort are blindzeln.
COBRA 11.x will run as demo for iIrc. 30 minutes. So you can test it. Klicktel is accessible with COBRA.

Yours sincerely

Arno
 

Von: Adriani90
Gesendet: Samstag, 6. April 2024 23:25
An: nvaccess/nvda
Cc: jammerhexe; Mention
Betreff: Re: [nvaccess/nvda] Access to German phone book Klicktel (#6135)

This is still an issue with clicktel 2024, NVDA cannot really read anything on its interface except for confirmation dialogs and some of the edit fields. With object navigation it is possible to explore some of the controls, though most of them do not have any accessible labels and you cannot route the mouse to them.
Jaws doesn't work with this interface at all, its virtual jaws cursor can read some of the controls but it is not possible to work with it.
Narator seems to be the only screen reader that recognizes properly some of the controls via its scan mode, but it is still far from an accessible UX.
The menu and the toolbar are not accessible at all with any of the screen readers.
We didn't read any user requests for clicktel accessibility since many years, neither in the german communication channels. It might be better to contact the company directly and to start an advisory initiative if they want. There is nothing NVDA can do at this stage.
I am closing this particular issue as can't fix.

**Comment by Adriani90** [comment]
Thanks Arno for this valuable update. It is certainly more than worth it to investigate how Cobra can work with it while all other screenreaders don‘t. I wonder if Cobra has some sort of custom app modules?
Is there a way to somehow send the code to NV Access so they can investigate?
I am reopening this one.

**Comment by jammerhexe** [comment]
Hi Adriani,

as I wrote in my last mail Baum is out of business. Of course the people there are still alive. But as far as I know they are no longer willing in giving any support to COBRA.
Jürgen Thelen who wrote the scrips for MY COBRA INSTALLATION to made it accessible FOR SAP is now employed by Papenmeyer 
https://www.google.com/search?q=j%C3%BCrgen+thelen+papenmeyer&sca_esv=ae442b666983332b&sxsrf=ACQVn09KcRwMZiSfm2CA3oUVwQvdlZEudg%3A1712492951348&source=hp&ei=l5ESZpP3EvKM7NYPzvK_-Ag&iflsig=ANes7DEAAAAAZhKfp-HLX64ce-C_BV27LyGjNuZzqyFf&ved=0ahUKEwjTnqywjbCFAxVyBtsEHU75D48Q4dUDCBc&uact=5&oq=j%C3%BCrgen+thelen+papenmeyer&gs_lp=Egdnd3Mtd2l6Ihlqw7xyZ2VuIHRoZWxlbiBwYXBlbm1leWVyMgUQIRigATIFECEYoAFIzooBUABY2YcBcAB4AJABAJgB4iOgAf65AaoBAzktNrgBA8gBAPgBAZgCBqAC2roBwgIFEAAYgATCAgYQABgWGB7CAgcQIRgKGKABmAMAkgcFOC0yLjSgB4QX&sclient=gws-wiz&zx=1712493112232&no_sw_cr=1#ip=1

Some other members of the Baum crew, too. Steffen König who was member of the support team is afaIk. Retired. I don’t know his private address.

Yours sincerely

Arno

	

Von: Adriani90
Gesendet: Sonntag, 7. April 2024 14:08
An: nvaccess/nvda
Cc: jammerhexe; Mention
Betreff: Re: [nvaccess/nvda] Access to German phone book Klicktel (#6135)

Thanks Arno for this valuable update. It is certainly more than worth it to investigate how Cobra can work with it while all other screenreaders don‘t. I wonder if Cobra has some sort of custom app modules?
Is there a way to somehow send the code to NV Access so they can investigate?
I am reopening this one.

**Comment by jammerhexe** [comment]
Hi Adriani,

today I asked Steffen König about his knowledge about the accessibility of Klicktel with COBRA. Below I copy his answer in German to my request. Probably it is helpful to resolve the problem.


Hallo Herr Schuh, 

unter Virgo 4 war die Klicktel mittels Script zugänglich gemacht worden,
wobei dieses Script im Wesentlichen die Verfolgung der Vorder- und/oder
Hintergrundfarbe des Auswahlbalkens in den verschiedenen Ergebnislisten der
Suche realisiert hat. 

Mit der Fusion von Virgo 4 und Blindows wurde in COBRA dann die unter
Blindows bereits vorhandene Möglichkeit genutzt, eine Anwendung mittels
Class.ini und Colors.ini zugänglich zu machen, indem die in der OSM.dll (OSM
steht für OffScreen Model) definierten Windows Controls, deren Eigenschaften
und Methoden von Anwendungen für die jeweiligen Fensterklassen angepasst
wurden. 

Kurz gesagt bedeutet dies, dass man mit Hilfe der von MS zur Verfügung
gestellten Accessibility Tools (wie z.B. dem Inspect) die Elementtypen einer
Anwendung, wie z.B. die Menubar, das Submenu, die Checkbox, der Radiobutton,
die Listbox, das Treeview etc. sowie deren Events, analysiert und sofern
noch nicht vorhanden in der OSM.dll verfügbar machen und dann via Class.ini
der Anwendung x zuordnen konnte. 

Auf diese Weise war es unter COBRA möglich den von einer Anwendung x
verwendeten Elementtypen, welche nicht dem Windows Standard entsprachen, via
Class.ini eine entsprechende Fensterklasse zuzuweisen, so dass COBRA dann
der jeweiligen Fensterklasse entsprechendes Verhalten an den Tag legte. 

So wurde beispielsweise den in der Klicktel nicht Windows-konformen Controls
via Class.ini verklickert, dass, wenn beispielsweise ein Element der Menubar
den Fokus hat, COBRA diesen verfolgt und entsprechend dem Windows Standard
behandelt, was zur Folge hatte, dass COBRA diesen Elementtyp als Hauptmenü
erkannt und sich entsprechend der Fokusevents etc. korrekt verhalten hat. 

Eigentlich gehe ich fest davon aus, dass sich auch die Entwickler von NVDA
der Microsoft Accessibility Tools bedienen. Dennoch hier ein Link, welcher
Ihnen mehr zu diesem Thema vermitteln sollte. 

<https://learn.microsoft.com/en-us/windows/win32/winauto/inspect-objects>

VG, Steffen König. 


-----Ursprüngliche Nachricht-----
Von: Arno Schuh ***@***.***> 
Gesendet: Donnerstag, 11. April 2024 11:03
An: ***@***.***
Betreff: p.m. Cobra und Klicktel

Hallo Herr König,

auch auf die Gefahr hin, dass Sie nichts mehr zu Cobra sagen können oder
wollen - ich habe da eine Frage:
Ich nutze aus verschiedenen Gründen immer noch gerne die Klicktel-CD, um
einfach Adressen zu kopieren usw.
Cobra hatte mit der Klicktel ja kein Problem, d. h., die Menüs, die Listen
usw. waren und sind mit Cobra zugänglich.
Ich hatte bei NVDA ein Ticket eingereicht und darum gebeten, die Klicktel
auch für NVDA zugänglich zu machen.
Leider sehen sie sich außer Stande dazu.

Wie hat Cobra das gemacht? Wurde dafür ein Script geschrieben? Können Sie
mir etwas an Informationen dazu geben, was ich an die Entwickler von NVDA
weitergeben könnte, damit sie testen können, ob NVDA zugänglich gemacht
werden kann?

Freundliche Grüße

Arno Schuh




Von: Adriani90
Gesendet: Sonntag, 7. April 2024 14:08
An: nvaccess/nvda
Cc: jammerhexe; Mention
Betreff: Re: [nvaccess/nvda] Access to German phone book Klicktel (#6135)

Thanks Arno for this valuable update. It is certainly more than worth it to investigate how Cobra can work with it while all other screenreaders don‘t. I wonder if Cobra has some sort of custom app modules?
Is there a way to somehow send the code to NV Access so they can investigate?
I am reopening this one.

**Comment by Adriani90** [comment]
@jammerhexe this seems like a very dirty workaround due to lack of accessibility. Did you try to contact 11880 Internet Services AG to talk with them about the software directly?
Given the fact that in Germany we soon have the new accessibility strengthening law (BFSG) it should give enough push for them to make their application more accessible.

**Comment by Adriani90** [comment]
@jammerhexe do you still have the issue with NVDA 2025.3.2 or last 64bit alpha and clicktel 2025 last update?
